jwcrypto/jwe.py

```diff
@@ -542,6 +600,25 @@ def _get_alg_enc_from_headers(self, jh):
         enc = self._jwa(encname)
         return alg, enc
 
+    def _encrypt(self, alg, enc, jh):
+        aad = base64url_encode(self.objects.get('protected', ''))
+        if 'aad' in self.objects:
+            aad += '.' + base64url_encode(self.objects['aad'])
```

Comment:
Isn't aad already encoded? Unless I'm mistaken, I think you're double-encoding here.